templates/node/events/index.js

```diff
@@ -13,37 +15,22 @@ const { CloudEvent, HTTP } = require('cloudevents');
  *  -H'Ce-specversion: 1.0' \
  *  http://localhost:8080
  *
- * The event data is extracted from the incoming event and provided as the first
- * parameter to the function. The CloudEvent object itself may be accessed via the
- * context parameter, For example:
- *
- * const incomingEvent = context.cloudevent;
- *
  * @param {Context} context the invocation context
- * @param {Object} user the CloudEvent data. If the data content type is application/json
- * this will be converted to an Object via JSON.parse()
+ * @param {Object} event the CloudEvent 
  */
-function verifyUser(context, user) {
-  if (!context.cloudevent) {
-    return {
-      message: 'No cloud event received'
-    };
-  }
+function handle(context, event) {
 
-  context.log.info('Processing user', user);
-  context.log.info(`CloudEvent received: ${context.cloudevent.toString()}`);
+  context.log.info("context");
+  console.log(JSON.stringify(context, null, 2));
+
+  context.log.info("event");
+  console.log(JSON.stringify(event, null, 2));
 
-  user = verify(user);
   return HTTP.binary(new CloudEvent({
-    source: 'function.verifyUser',
-    type: 'user:verified',
-    data: user
+    source: 'event.handler',
+    type: 'echo',
+    data: event
   }));
```

Comment:
So with an eye on our related conversation about accepting an actual CloudEvent named _event_ as the second parameter, it may gain some support from its effect on the default implementation.  If we assume that the default handler echoes a cloud event, this cleans up nice:
```
function handle(context, event) {
  return event
}
```

Reply:
That absolutely appeals to my aesthetic!! :)

However, I kinda dig showing a bit of the `HTTP` library, and I think it's important to set the source & type fields on the returned event.

It might be more useful to the new `func` user to see `data: event.data` though, just to reinforce that the event is an actual `CloudEvent`.  But I'd rather not do that until after the runtime is changed to do that. As it stands, this template will work with either the current or the future runtime.

Reply:
Also, if the event source is a broker, won't returning the event result in an infinite loop?

Reply:
> Also, if the event source is a broker, won't returning the event result in an infinite loop?

That's a good question. I suppose only if the function is subscribed via Trigger, and only filtering on event type - and even then, that assume's the broker is not smart enough to deal with this.

Reply:
> However, I kinda dig showing a bit of the `HTTP` library, and I think it's important to set the source & type fields on the returned event.

```js
function handle(context, event) {
  return event.cloneWith({ source: 'echo.fn', type: 'echo' });
}
```

Reply:
>  But I'd rather not do that until after the runtime is changed to do that.

If this lands, https://github.com/boson-project/faas-js-runtime/pull/96 - will you coordinate with this PR? ([TypeScript](https://github.com/boson-project/func/pull/355) needs to change as well).

Reply:
> If this lands, [boson-project/faas-js-runtime#96](https://github.com/boson-project/faas-js-runtime/pull/96) - will you coordinate with this PR? ([TypeScript](https://github.com/boson-project/func/pull/355) needs to change as well).

Sure, but... define "coordinate" in this context. I think they can each be merged independently. Or I can wait until after #355  lands and adjust those templates, too?

Reply:
@jcrossley3 I mean, primarily, that this PR is updated to reflect those changes. Due to the fact that these are user facing API changes, we'll also want to update docs/guides/nodejs.md.

Reply:
Ah, gotcha. I just pushed a commit that updates the nodejs guide.